include/nuttx/sched_note.h

```diff
@@ -391,27 +423,62 @@ ssize_t sched_note_get(FAR uint8_t *buffer, size_t buflen);
 ssize_t sched_note_size(void);
 #endif
 
+/****************************************************************************
+ * Name: sched_note_buffer_clear
+ *
+ * Description:
+ *   Clear all contents of the circular buffer.
+ *
+ * Input Parameters:
+ *   oneshot - Oneshot mode flag.  If true, the buffer put operation stops
+ *             when whole buffer is used.
+ *
+ * Returned Value:
+ *   None.
+ *
+ ****************************************************************************/
+#ifdef NOTE_FEATURE_BUFFER_REWINDABLE
+void sched_note_buffer_clear(int oneshot);
```

Comment:
Why we expose the internal function?

Reply:
It is because the function is in sched_note_buffer.c because it manages the note buffer, but the trace start/stop function in sched_note.c need it.
The reason of sched_note_buffer_rewind() is same, too. (I also think it is not good design...)

The another idea for it is:
- Provide the struct which inherits  lib_outstream_s. It contains the function pointers to manage the buffer (clear/rewind).
- sched_note_buffer.c sets the function pointer of sched_note_buffer_clear/rewind() to the new struct.
- sched_note.c calls the function if the pointer is not NULL.